### DISCUSSION TITLE
Modify source code to autoboot script

### DISCUSSION THREAD

**Discussion by Richie-lem** [suchmememanyskill/TegraExplorer#94] [Ideas]
Hi,
My idea is:
Based on this project https://github.com/i-lost-my-bagel/AtmosphereFolderSwitch, I want to compile a new TegraExplorer payload that only boot a specific script before booting atmosphere payload

I want to have two version of this new payload, both of them boot a specific script, one for the sysnand and one for the emunand, then start atmosphere

By this way we can have two different atmosphere folder and scripts booted before atmosphere payload just rename atmosphere folders for emunand or sysnand

I can do it myself, so my question is: Is there any risk by editing your source code and boot it on my own switch ?

**Comment by suchmememanyskill** [comment]
You may be able to re-use the functionality that loads startup.te on payload boot. In here you could make a menu to 'boot' sys or emu.

Although nothing is stopping you from editing the source

**Reply by Richie-lem** [reply]
Thank you so much for your quick response =)

_You may be able to re-use the functionality that loads startup.te on payload boot. In here you could make a menu to 'boot' sys or emu._
Yes this is exacly what I want to do, I already found how to do it

But is there a risk to brick my switch if my code fail for exemple ? What happened if I got an error directly while using the payload ?
Is there some good way to test safely ?

**Reply by suchmememanyskill** [reply]
For launching a payload? Why would that brick?

Worst case the console crashes.

There's no real way to test scripts outside of TE. There's a PC emulator but it's heavily flawed
